From: Rene van Paassen <Rene.vanPaassen@gmail.com>
Date: 2014-03-11 16:34:05 +0100
Subject: gcc on sles fails to emit template instantiation
References: 
Upstream: no

Added explicit template instantiation for mesh generator, needed for gcc on
sles


Index: sinbad-ogre-dd30349ea667/OgreMain/src/OgreProgressiveMeshGenerator.cpp
===================================================================
--- sinbad-ogre-dd30349ea667.orig/OgreMain/src/OgreProgressiveMeshGenerator.cpp
+++ sinbad-ogre-dd30349ea667/OgreMain/src/OgreProgressiveMeshGenerator.cpp
@@ -257,6 +257,18 @@ void ProgressiveMeshGenerator::addIndexD
 	}
 }
 
+template
+void ProgressiveMeshGenerator::addIndexDataImpl<unsigned short>(unsigned short* iPos,
+				      const unsigned short* iEnd,
+				      VertexLookupList& lookup,
+				      unsigned short submeshID);
+template
+void ProgressiveMeshGenerator::addIndexDataImpl<unsigned int>(unsigned int* iPos,
+				    const unsigned int* iEnd,
+				    VertexLookupList& lookup,
+				    unsigned short submeshID);
+
+
 void ProgressiveMeshGenerator::addIndexData(IndexData* indexData, bool useSharedVertexLookup, unsigned short submeshID)
 {
 	const HardwareIndexBufferSharedPtr& ibuf = indexData->indexBuffer;
